pkg/apis/pipeline/v1/pipeline_validation.go

```diff
@@ -101,8 +101,8 @@ func (ps *PipelineSpec) ValidateBetaFields(ctx context.Context) *apis.FieldError
 	}
 	// Indexing into array parameters
 	arrayParamIndexingRefs := ps.GetIndexingReferencesToArrayParams()
-	if len(arrayParamIndexingRefs) != 0 && !config.CheckAlphaOrBetaAPIFields(ctx) {
-		errs = errs.Also(apis.ErrGeneric(fmt.Sprintf("cannot index into array parameters when 'enable-api-fields' is 'stable', but found indexing references: %s", arrayParamIndexingRefs)))
+	if len(arrayParamIndexingRefs) != 0 {
```

Comment:
I think we might still need to check the enabled api fields in the condition statement?

```suggestion
	if len(arrayParamIndexingRefs) != 0  &&  version.ValidateEnabledAPIFields(ctx, "indexing into array parameters", config.BetaAPIFields) != nil {
```

Reply:
I don't think this is necessary; if version.ValidateEnabledAPIFields returns an error, this error will be bubbled up through `ValidateBetaFields`. There are similar calls to version.ValidateEnabledAPIFields within this function, and testing beta feature flags for array indexing still passes: https://github.com/tektoncd/pipeline/blob/3364c7269be58898f1466174ba0cc3f2a04b9ece/pkg/apis/pipeline/v1/pipeline_validation_test.go#L3358-L3370

Reply:
Thanks for the clarifications. On first thought I was trying to keep the previous 'semantics'(sorry should have put a NIT in the comment), makes perfect sense to me on this change logically and after looking at how other version.ValidateEnabledAPIFields i think this is the ideal way to go.